packages/eas-cli/src/project/android/applicationId.ts

```diff
@@ -33,6 +33,57 @@ export async function ensureApplicationIdIsDefinedForManagedProjectAsync(
   }
 }
 
+export class AmbiguousApplicationIdError extends Error {
+  constructor(message?: string) {
+    super(message ?? 'Could not resolve applicationId.');
+  }
+}
+
+export async function getApplicationIdFromBareAsync(
+  projectDir: string,
+  gradleContext?: GradleBuildContext
+): Promise<string> {
+  const errorMessage = 'Could not read application id from Android project.';
+
+  if (gradleContext) {
+    const buildGradle = await gradleUtils.getAppBuildGradleAsync(projectDir);
+    const applicationIdSuffix = gradleUtils.resolveConfigValue(
+      buildGradle,
+      'applicationIdSuffix',
+      gradleContext.flavor
+    );
+    if (applicationIdSuffix) {
+      throw new Error('applicationIdSuffix is not supported');
+    }
+    const applicationId = gradleUtils.resolveConfigValue(
+      buildGradle,
+      'applicationId',
+      gradleContext.flavor
+    );
+    return nullthrows(applicationId, errorMessage);
+  } else {
+    // should return value only if productFlavors are not used
+    let buildGradlePath = null;
+    try {
+      buildGradlePath = AndroidConfig.Paths.getAppBuildGradleFilePath(projectDir);
+    } catch {}
+    if (!buildGradlePath || !(await fs.pathExists(buildGradlePath))) {
+      throw new Error(errorMessage);
+    }
+    const buildGradle = await fs.readFile(buildGradlePath, 'utf8');
+    const matchResult = buildGradle.match(/applicationId ['"](.*)['"]/);
+    if (buildGradle.match(/applicationIdSuffix/)) {
+      throw new Error('"applicationIdSuffix is not supported');
+    }
+    if (buildGradle.match(/productFlavors/)) {
+      throw new AmbiguousApplicationIdError(
+        'Failed to resolve applicationId in multi-flavor project'
```

Comment:
Please decide on using or not using dots in error messages. This comment applies to the whole PR.